gateway/requests/requests.go

```diff
@@ -90,3 +90,20 @@ type AsyncReport struct {
 type DeleteFunctionRequest struct {
 	FunctionName string `json:"functionName"`
 }
+
+// CreateSecretRequest create a secret w/ annotations
+type CreateSecretRequest struct {
+	Secret Secret `json:"secret"`
+}
+
+// DeleteSecretRequest remote a secret by name
+type DeleteSecretRequest struct {
+	SecretName string `json:"secretName"`
+}
+
+// Secret schema use Value only in write-only http verbs
+type Secret struct {
+	Name        string             `json:"name"`
+	Value       string             `json:"value"` // write-only
+	Annotations *map[string]string `json:"annotations"`
```

Comment:
I think annotations or metadata will be useful and necessary. Do both Swarm and Kubernetes support adding this to secrets?

Reply:
Looking at this again, perhaps we should move it to the CreateSecretRequest struct instead?

Reply:
Why only on the Create? Won't we want to be able to read and display those annotations? 

Is there a reason to use a pointer here? `maps` behave like pointers already, e.g. 
```golang
var m map[string]string
m = nil
fmt.Printf("%d", len(m))
```
is valid and I think when you json unmarshal and the value is empty or omitted, the map will be set nil as well.

Reply:
It might be due to copy/paste from other areas of code? We should add omitempty on some of these properties too.

Reply:
How about simply a `SecretRequest` type? Members could be:
```
Secret Secret `json:"secret"`
Annotations map[string]string `json:"annotations"`
```
POST, PUT are pretty straightforward. GET would then return `[]SecretRequest` instead, omitting certain values ;).

Reply:
I would highly recommend separating the Request struct from the Response struct.  It becomes much easier to document and to extend in the future.